oss-internship-2020/libuv/callbacks/callbacks.cc

```diff
@@ -0,0 +1,93 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "callbacks.h"
+
+#include <iostream>
+
+size_t iterations = 0;
+size_t constexpr kMaxIterations = 1'000'000;
+
+// Stop the handle if the methods was called kMaxIterations times
+void IdleCallback(uv_idle_t* handle) {
+  iterations++;
+  if (iterations > kMaxIterations) {
+    std::cout << "IdleCallback was called " << kMaxIterations << " times"
+              << std::endl;
+    uv_idle_stop(handle);
+  }
+}
+
+static char buffer[1024];
+static uv_buf_t iov;
```

Comment:
Move those up to the other global `iterations` variable and also prefix with `g_`.